clickanalysis/middleware/__init__.py

```diff
@@ -10,32 +13,22 @@ class ClickAnalysisMiddleware(object):
     """
 
     def process_request(self, request):
+
         if request.GET.get('cid', ''):  # Will process the logic only if campaign id is present
             try:
                 campaign = Campaign.objects.get(campaign_id=request.GET['cid'])
-            except:
+            except Campaign.DoesNotExist:
                 campaign = None
 
             # Clearing the cid parameter from the link or providing a clean url for user to redirect
-            get_params = request.get_full_path().split('?')[1].split('&')
-            params_to_send = []
-            for g_p in get_params:
-                name = g_p.split('=')[0]
-                if name != 'cid':
-                    params_to_send.append(g_p)
-
-            params = '&'.join(params_to_send)
-            if params_to_send:
-                url = request.get_full_path().split('?')[0] + '?' + params
-            else:
-                url = request.get_full_path().split('?')[0]
-
-            if campaign:  # If campaign was found then here the click will be recorded
+            url_parsed = urlparse(request.get_full_path())
+            get_params = parse_qsl(url_parsed.query)
+            params_to_send = urlencode([(k, v) for k, v in get_params if k != 'cid'])
+            url = urlunsplit(url_parsed[:3] + (params_to_send,) + url_parsed[5:])
+
+            if campaign is not None:  # If campaign was found then here the click will be recorded
```

Comment:
`None` is a singleton in Python, with singleton it is recommended to use `is` rather than a plain `==` or simple `if`(Second point [PEP-008: Programming Recommendations](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)). A simple `if` checks if the item is a [truthy value](https://docs.python.org/2/library/stdtypes.html#truth-value-testing), which results in a chain of calls to `__bool__`, `__len__` on object on the other and `is` directly compares items by identity.

`True` and `False` are singletons too, and in CPython 2.6+ almost all occurrences of `x == True` etc have [been already converted](https://hg.python.org/cpython/rev/01a7e66525c2/) to `x is True`.